src/main/scala/symsim/ValueFunction.scala

```diff
@@ -0,0 +1,61 @@
+package symsim
+
+import cats.Monad
+import cats.syntax.functor.*
+import cats.syntax.flatMap.*
+import cats.syntax.foldable.*
+import cats.syntax.option.*
+import org.scalacheck.Gen
+import org.typelevel.paiges.Doc
+import symsim.concrete.{ConcreteExactRL, Randomized}
+
+
+trait ValueFunction[State, ObservableState, Action, Reward, Scheduler[_]]:
+   type VF
+   def bestAction (vf: VF) (s: State): Action
+   def chooseAction (vf: VF) (s: State): Scheduler[Action]
+   def initialize: VF
+
+
+trait QTable[State, ObservableState, Action, Reward, Scheduler[_]]
+   extends ValueFunction[State, ObservableState, Action, Reward, Scheduler]:
+   this: ExactRL[State, ObservableState, Action, Reward, Scheduler] =>
+   import agent.instances.*
+
+   type VF = Map[ObservableState, Map[Action, Reward]]
```

Comment:
change to `type Q = Map[ObservableState, Map[Action, Reward]]` and change the next line to `type VF = Q` (I think this gives a more natural presentation, despite being equivalent).